projects/igniteui-angular/src/lib/combo/combo.component.ts

```diff
@@ -1107,22 +1107,24 @@ export class IgxComboComponent extends DisplayDensityBase implements IgxComboBas
         if (this.groupKey) {
             Object.assign(addedItem, { [this.groupKey]: this.defaultFallbackGroup });
         }
-        const oldCollection = this.data;
-        const newCollection = [...this.data];
-        newCollection.push(addedItem);
+        // do not allow data to be mutated through event args
```

Comment:
one can still mutate the data if it is array of objects by changing the objects' internal properties. If you do need to prevent mutation provide deep copy here

Reply:
You can still mutate the objects, which is fine, since you already have access to the `data` that the combo is bound to.
I meant that the `data` array itself cannot be mutated through the event args. Previously, as it was `oldCollection = this.data`, some could do `event.oldCollection.push(obj)` and that would mutate `combo.data` - it will no longer do that.